### PR TITLE
BREAKING CHANGES: Theme default props

### PR DESCRIPTION
**Pull request by vinicius33**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [x] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

Currently, we aren't using any fallback as a default theme for the other applications other than `rooster`. **What we are providing through this PR is an extension an deep merge of the `StandardTheme` regardless if Theme has been specified or not**. 

The idea from now on is to import the `ThemeProvider` from Cardinal like so:

```javascript
	<CardinalThemeProvider theme={foo|bar|null|undefined}>
 		...
    </CardinalThemeProvider>
```

Under the hood, it's going to merge the themes and in case of an undefined theme, it's going to assume the [`StandardTheme`](https://github.com/foodora/logistics-cardinal-ui/pull/110/files#diff-9e8f4f512f3bd081882e6d1d1bce9fc7).

Also, what I've thought is to avoid namespacing nested properties. I think it's sensible to stick to a pattern like bootstrap does on their `variables.scss`.

I didn't come up with this pattern 👍 . It's indeed recommended by `styled-components` 💅 [https://github.com/styled-components/styled-components-experimentation/blob/master/component-libraries/shared-component-libraries.md](https://github.com/styled-components/styled-components-experimentation/blob/master/component-libraries/shared-component-libraries.md)

**The PR fulfills these requirements:**

- [x] All tests are passing
- [x] New/updated tests are included
- [x] Run the precommit hook (it's going to run all required actions such as `Prettier`, `Eslint`, `Jest` and so on...)

**Other information:**
* Removed usage of bootstrap and react-bootstrap 🎉 
* Refactored <Button /> component to avoid using bootstrap.
* Avoiding extending `Pure HTML Components` via `Styled-Components` once its usage has been dropped on v4.
* Removed unused files and folders

Please, don't hesitate to make improvements, ask questions or even request changes. 

Thanks,

### CODE REVIEW

File: .storybook/decorators.js [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
 
-  // TODO Remove the following condition when we added a bootstrap for hurrier
-  if (selectedTheme === 'rooster') {
-    require(`../src/themes/${selectedTheme}/bootstrap/css/bootstrap.css`);
-  }
+  createGlobalStyles(themes[selectedTheme]);
 
   return (
-    <ThemeProvider theme={themes[selectedTheme]}>
+    <CardinalThemeProvider theme={themes[selectedTheme]}>
```
Comment (vinicius33) [inline .storybook/decorators.js]: ^ important

File: src/themes/hurrier.js [hunk 2]
```diff
@@ -76,24 +81,260 @@ const borderRadius = {
...
+
+const boxShadow = {
+  switchHandler: '0px 0px 1px 5px rgba(255, 255, 255, 0.2)',
+  drawer: '0 2px 34px 0 rgba(0, 0, 0, 0.3)',
+  inputButton: 'inset 0 1px 1px rgba(0, 0, 0, 0.075)',
 };
 
-export default {
+// Required Fields
+export const aliases = {
```
Comment (vinicius33) [inline src/themes/hurrier.js]: From now on we only care about this object and below. 
If you are going to reuse objects such as colors, fontWeights, fontSizes it's totally up to you.

File: src/themes/standard.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
+// Required Fields
+export const aliases = {
```
Comment (vinicius33) [inline src/themes/standard.js]: Any new property has to be included into `StandardTheme` before any other `Theme`. That's the compliance.

File: src/themes/global-styles.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { injectGlobal } from 'styled-components';
+import 'normalize.css';
+
+export const createGlobalStyles = (theme) => injectGlobal`
```
[the remark this answers is not in the document]
Comment (vinicius33) [reply src/themes/global-styles.js]: never mind about this one. We'll call it on Provider.js

**Comment by Tagir-A** [comment]
So, as you explained this PR to me, everything looks fine
I totally support defaults :)

File: src/themes/hurrier.js (hunk 2 above)
Comment (Tagir-A) [reply src/themes/hurrier.js]: I'll need an introduction from you on how these aliases are working in real life

**Pull request merged**